dissect/target/plugins/scrape/qfind.py

```diff
@@ -109,9 +133,21 @@ def qfind(
                 tmp[re.compile(re.escape(encoded_needle), re.IGNORECASE)] = _
             needle_lookup = tmp
 
+        if regex:
+            tmp = {}
+            for needle, _ in needle_lookup.items():
+                tmp[re.compile(needle, re.IGNORECASE if ignore_case else re.NOFLAG)] = _
+            needle_lookup = tmp
+
+        # Output records if invoked through target-query only.
+        # TODO: Find a more suitable method to derive invocation method.
+        record = False
+        if "target-query" in sys.argv[0]:
+            record = True
```

Comment:
I had thought about this use-case while writing this plugin but wanted to get this working normally first and then worry about it later 😄.

My initial gut feeling is to make the plugin be record-yielding first and foremost, and move all the "graphical" stuff to the `target-qfind` `main`. You could deal with the progress bar by just having a `progress=True` keyword argument that is not exposed with an argument.

Another option is to make most of this method a `_qfind` method in the plugin, and have a small wrapping `qfind` record `@export`, and an `@internal` `qfind_ext`. Where you would use `qfind_ext` in `target-qfind`.

I think I like the first option a bit better, but it's good to have an open discussion about it to exchange ideas.